### PR TITLE
V1.13.x webhook deletes

### PR DESCRIPTION
**Pull request by sheidkamp**
# Description
Backport of https://github.com/solo-io/gloo/pull/8282
And pulling in https://github.com/solo-io/gloo/pull/7539 so tests pass
# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/main/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: .github/workflows/regression-tests.yaml [hunk 1]
```diff
@@ -1,5 +1,17 @@
...
+    branches:
+      - 'master'
+  pull_request:
+
+env:
+  VERSION: '1.0.0-ci'
+  HELM_VERSION: 'v3.6.0'
+  KUBECTL_VERSION: 'v1.22.4'
+  CLUSTER_NODE_VERSION: 'v1.22.4@sha256:ca3587e6e545a96c07bf82e2c46503d9ef86fc704f44c17577fca7bcabf5f978'
+  GITHUB_TOKEN: ${{ secrets.PERSONAL_ACCESS_TOKEN }}
```
Comment (sam-heilbron) [inline .github/workflows/regression-tests.yaml:13]: I thnk a lot of this came from the cherry-pick and can be removed

File: test/kube2e/README.md [hunk 2]
```diff
@@ -19,45 +19,34 @@ If a test fails, you can retry it from a [browser window](https://docs.github.co
...
 [ci/deploy-to-kind-cluster.sh](`https://github.com/solo-io/gloo/blob/master/ci/deploy-to-kind-cluster.sh`) gets run in CI to setup the test environment for the above requirements.
 It accepts a number of environment variables, to control the creation of a kind cluster and deployment of Gloo resources to that kind cluster.
 
-| Name                  | Default    | Description |
-| ---                   |   ---      |    ---      |
-| CLUSTER_NAME          | kind       | The name of the cluster that will be generated |
-| CLUSTER_NODE_VERSION  | v1.24.7    | The version of the [Node Docker image](https://hub.docker.com/r/kindest/node/) to use for booting the cluster |
-| VERSION               | 0.0.0-kind1 | The version used to tag Gloo images that are deployed to the cluster |
-| KUBE2E_TESTS          | gateway    | Name of the test suite to be run. Options: `'gateway', 'gloo', 'ingress', 'helm', 'gloomtls', 'glooctl'` |
+| Name                 | Default  | Description                                                                                                         |
```
Comment (sam-heilbron) [inline test/kube2e/README.md:32]: I think we can back these changes out. They imply we changed defaults to our setup kind script (again, pulled from the cherry-pick), but that isn't true for this branch

File: test/kube2e/util.go [hunk 3]
```diff
@@ -288,12 +285,6 @@ func ToFile(content string) string {
 	return f.Name()
 }
 
-// PatchResource mutates an existing resource, retrying if a resourceVersionError is encountered
```
Comment (sam-heilbron) [inline test/kube2e/util.go:291]: We can delete this, though I think it's easiest to leave it as deprecated. If we delete, we'll need to ensure the solo-projects changes account for this

Comment (sheidkamp) [reply test/kube2e/util.go:291]: restoring

File: test/kube2e/README.md (hunk 2 above)
Comment (sheidkamp) [reply test/kube2e/README.md:32]: The updated default matches what is in the `deploy-to-kind-cluster.sh` script (unchanged in this PR), so I think we should keep this

File: .github/workflows/regression-tests.yaml (hunk 1 above)
Comment (sheidkamp) [reply .github/workflows/regression-tests.yaml]: Removed

**Comment by sheidkamp** [comment]
/kick (maybe not a flake, maybe due to changes) https://storage.googleapis.com/solo-public-build-logs/logs.html?buildid=65b08da8-92c7-431f-82c5-1b2f1da992f8

**Review by jackstine — APPROVED** [review]

**Pull request merged**